### PR TITLE
E2E-monitor server change

### PR DESCRIPTION
**Pull request by naemono**
## What is this change?
* This primarily updates our e2e tests to use a new e2e-monitor server defined in a new vault path, as our previous server is using an old k8s version and we thought it would be good to use our new offering as a test for this.
* This adds (back?) a missing variable needed when running our e2e tests `E2E-IMG` which is used here: https://github.com/elastic/cloud-on-k8s/blob/20f11e756989d9811d3f22f638949a81409a4ef6/Makefile#L443
* Migrates E2E tests to use an `api_key` instead of a `secret` for integrating with APM Server, which is required in our new offering.
* Updates the version of Elastic Agent we are using in our monitoring of e2e tests.

## TODO
- [x] Test locally with a limited set of e2e tests `make e2e-run  E2E_STACK_VERSION=8.9.0 TESTS_MATCH=TestFleetKubernetesNonRootIntegrationRecipe`
- [ ] Run a full set of e2e tests via our CI Pipeline

### CODE REVIEW

**Comment by naemono** [comment]
buildkite test this -f s=8.9.0,p=gke

**Comment by thbkrkr** [comment]
> This adds (back?) a missing variable needed when running our e2e tests E2E-IMG which is used here:

I went a little too far with the cleaning in #6959. It can be useful if you want 'make e2e-run` directly. In CI, it is set dynamically by the pipeline generator using the image was built during the build:

https://github.com/elastic/cloud-on-k8s/blob/20f11e756989d9811d3f22f638949a81409a4ef6/.buildkite/e2e/pipeline-gen/main.go#L377

**Review by thbkrkr — APPROVED** [review]

File: config/e2e/monitoring.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ metadata:
   name: e2e-agent
   namespace: {{ .E2ENamespace }}
 spec:
-  version: 8.4.0 # needs to less or equal to e2e-monitor version
+  version: 8.9.0 # needs to less or equal to e2e-monitor version
```
Comment (thbkrkr) [inline config/e2e/monitoring.yaml:8]: Why not the latest 8.10.4?

Comment (naemono) [reply config/e2e/monitoring.yaml]: No good reason. Fixed.

**Comment by thbkrkr** [comment]
buildkite test this -f s=8.10.4,p=gke,t=TestSmoke

**Comment by naemono** [comment]
@pebrc @thbkrkr I know this is already approved, but I'm investigating some oddities from some of the data in the new e2e server (missing fields) so I'm upgrading k8s, and logs elastic-agent integration package (they were really old) to see if it helps resolve the issue. Just an FYI.

**Comment by naemono** [comment]
buildkite test this -f s=8.10.4,p=gke,t=TestSmoke

**Comment by naemono** [comment]
buildkite test this -f E2E_TAGS=es,p=gke

**Comment by naemono** [comment]
buildkite test this -f E2E_TAGS=es,p=gke

**Comment by naemono** [comment]
buildkite test this -f E2E_TAGS=agent,p=gke,t=TestFleetKubernetesNonRootIntegrationRecipe

**Comment by naemono** [comment]
buildkite test this -f E2E_TAGS=agent,p=gke,t=TestFleetKubernetesNonRootIntegrationRecipe

**Comment by naemono** [comment]
Ok, this can be merged after the builds finish. After adding additional permissions, I'm finally seeing the `kubernetes.container.*` data I (and the dashboard) is expecting...

**Pull request merged**